src/main/java/org/jabref/gui/FindUnlinkedFilesDialog.java

```diff
@@ -527,6 +535,88 @@ public void stateChanged(ChangeEvent e) {
         });
     }
 
+    /**
+     * This will start the export of all files of all selected nodes in this
+     * dialogs tree view. <br>
+     * <br>
+     * The export itself will run in a seperate thread, whilst this dialog will
+     * be showing a progress bar, until the thread has finished its work. <br>
+     * <br>
+     * When the export has finished, the {@link #exportFinishedHandler(java.util.List)} is
+     * invoked.
+     */
+    private void startExport() {
+
+        if (treeModel == null) {
+            return;
+        }
+        setDefaultCloseOperation(WindowConstants.DO_NOTHING_ON_CLOSE);
+
+        CheckableTreeNode root = (CheckableTreeNode) treeModel.getRoot();
+
+        final List<File> fileList = getFileListFromNode(root);
+
+        if ((fileList == null) || fileList.isEmpty()) {
+            return;
+        }
+
+        progressBarImporting.setVisible(true);
+        labelExportingInfo.setVisible(true);
+        buttonExport.setVisible(false);
+        buttonApply.setVisible(false);
+        buttonClose.setVisible(false);
+        disOrEnableDialog(false);
+
+        labelExportingInfo.setEnabled(true);
+
+        progressBarImporting.setMinimum(0);
+        progressBarImporting.setMaximum(fileList.size());
+        progressBarImporting.setValue(0);
+        progressBarImporting.setString("");
+
+        FileDialogConfiguration fileDialogConfiguration = new FileDialogConfiguration.Builder()
+                .withInitialDirectory(Globals.prefs.get(JabRefPreferences.WORKING_DIRECTORY)).build();
+        DialogService ds = new FXDialogService();
+
+        Optional<Path> exportPath = DefaultTaskExecutor
+                .runInJavaFXThread(() -> ds.showFileSaveDialog(fileDialogConfiguration));
+
```

Comment:
Please add a check if `exportPath` is present. If not (i.e. the user clicked cancel), then we should return early.